internal/controllers/envoyfleet_config_template.go

```diff
@@ -0,0 +1,77 @@
+// MIT License
+//
+// Copyright (c) 2022 Kubeshop
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package controllers
+
+var envoyConfigTemplate = `
+node:
+  cluster: %s
+
+dynamic_resources:
+  ads_config:
+    api_type: GRPC
+    transport_api_version: V3
+    grpc_services:
+    - envoy_grpc:
+        cluster_name: xds_cluster
+  cds_config:
+    resource_api_version: V3
+    ads: {}
+  lds_config:
+    resource_api_version: V3
+    ads: {}
+
+static_resources:
+  secrets:
+  - name: token
+    generic_secret:
+      secret:
+        inline_string: "Z6MX7NreJumWLmf6unsQ5uiEUrTBxfNtqG9Vy5Kjktnvfj-_fRCBO9EU1mL1YzAJ"
```

Comment:
harcoded secrets in a template?

Reply:
I've changed this to `<stub_token_secret>` because it needs to be specified so that the user can late adjust it. I really don't think it's a good idea to have them specify/paste:

```yaml
  secrets:
  - name: token
    generic_secret:
      secret:
        inline_string: "<some_token>"
```

Having them change one line is better.

What do you think @kylehodgetts?

Reply:
make it so. Replace the token with a place holder

Reply:
It's been replaced with `<stub_token_secret>` in `internal/controllers/envoyfleet_config_template.go` and in the `docs/docs/guides/oauth2.md` guide.